### ISSUE TITLE
[a11y]: ContainedList [with persistent search & expandable search] - label for search region announced 2x

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#13945]
### Package
@carbon/react

### Browser
Safari & FF & iPhone

### Package version
carbon/react@1.30.0

### Automated testing tool and ruleset
Manual QA

### Assistive technology
JAWS & VO

### Description
Label for the search region

Label "search" announced twice

### WCAG 2.1 Violation
2.4.6 Headings and Labels 

### Reproduction/example
Record `Number 33` in the below table:
https://airtable.com/shrFsaeHhImOkuVNR

<img width="1229" alt="Screenshot 2023-06-26 at 15 22 46" src="https://github.com/carbon-design-system/carbon/assets/32720851/847e2583-3ecf-47a1-a9bf-2cf253756ed5">

### Steps to reproduce
Any redundant, excessively wordy or extraneous announcement can be distracting for the screen reader users. 

https://react.carbondesignsystem.com/iframe.html?args=&id=components-containedlist--with-expandable-search&viewMode=story

### Recommended fix
Avoid redundant announcement

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by alisonjoseph** [comment]
This is actually an issue with the Search component.

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/searchbox_role

> Including role="searchbox" when the form is a search and the label indicates the element is a search may result in assistive technology announcing something along the lines of "search search this site search box", which is redundant. The inclusion of role="searchbox" is not necessary:

**Comment by andreancardona** [comment]
@alisonjoseph chatted with Gower and it seems a good solution would be to update the placeholder to `Filter` :)

**Comment by mbgower** [comment]
I also agree with Alison's quote from mozilla (and have posted this somewhere ? else) that the use of the searchbox role is often unnecessary.
